core/src/main/java/org/apache/calcite/rel/rules/SubQueryRemoveRule.java

```diff
@@ -235,9 +235,9 @@ private RexNode rewriteSome(RexSubQuery e, Set<CorrelationId> variablesSet,
       //   end as v
       // from emp as e
       // left outer join (
-      //   select max(deptno) as m, count(*) as c, count(deptno) as d,
+      //   select name, max(deptno) as m, count(*) as c, count(deptno) as d,
       //       "alwaysTrue" as indicator
-      //   group by name from emp) as q on e.name = q.name
+      //   from emp group by name) as q on e.name = q.name
```

Comment:
Thanks, there is no need to put `name` in the project list because it is already used as a group key.

BTW, this change is too small, can we merge it into a bigger one ?

Reply:
> Thanks, there is no need to put `name` in the project list because it is already used as a group key.
> 
> BTW, this change is too small, can we merge it into a bigger one ?
but, there isn't group key in implementation code

Reply:
The main idea of this change is to add a `name` in the select list, because `name` is referenced in the join condition.
Yes, this change seems too small, shall we hold ?

Reply:
I know what you're saying, @danny0405 - projecting `name` it isn't necessary in algebra because `Aggregate` always projects its group keys.

But it is necessary in SQL. The SQL as it stands is invalid, and fixing the SQL might help some people.

We recently had a discussion on dev@ about "small" PRs, and agreed to accept them. No JIRA case needed, the commit comment should make clear that it is a "small" change, no need to append the contributor's name, and the release manager should strip out such commits from the release notes.

I'm going to merge this one.